message/infrastructure/sql/subscriber.go

```diff
@@ -200,46 +214,70 @@ func (s *Subscriber) query(
 		}
 	}()
 
-	selectArgs, err := s.config.SchemaAdapter.SelectArgs(topic, s.config.ConsumerGroup)
-	if err != nil {
-		return errors.Wrap(err, "could not get args for the select query")
-	}
-
-	logger.Trace(selectQuery, watermill.LogFields{
-		"args": fmt.Sprintf("%+v", selectArgs),
+	selectQuery, selectQueryArgs := s.config.SchemaAdapter.SelectQuery(
+		topic,
+		s.config.ConsumerGroup,
+		s.config.OffsetsAdapter,
+	)
+	logger.Trace("Querying message", watermill.LogFields{
+		"query":      selectQuery,
+		"query_args": sqlArgsToLog(selectQueryArgs),
 	})
+	row := tx.QueryRowContext(ctx, selectQuery, selectQueryArgs...)
 
-	row := selectStmt.QueryRowContext(ctx, selectArgs...)
-
-	var offset int
-	var msg *message.Message
-	offset, msg, err = s.config.SchemaAdapter.UnmarshalMessage(row)
+	offset, msg, err := s.config.SchemaAdapter.UnmarshalMessage(row)
 	if errors.Cause(err) == sql.ErrNoRows {
 		// wait until polling for the next message
+		logger.Debug("No more messages, waiting until next query", watermill.LogFields{
+			"wait_time": s.config.PollInterval,
+		})
 		time.Sleep(s.config.PollInterval)
 		return nil
-	}
-	if err != nil {
+	} else if err != nil {
 		return errors.Wrap(err, "could not unmarshal message from query")
 	}
 
 	logger = logger.With(watermill.LogFields{
 		"msg_uuid": msg.UUID,
 	})
 
-	// todo: different acking strategies?
-	acked := s.sendMessage(ctx, msg, out, logger)
-	if acked {
-		var ackArgs []interface{}
-		ackArgs, err = s.config.SchemaAdapter.AckArgs(offset, s.config.ConsumerGroup)
+	consumedQuery, consumedArgs := s.config.OffsetsAdapter.ConsumedMessageQuery(
+		topic,
+		offset,
+		s.config.ConsumerGroup,
+		s.consumerIdBytes,
+	)
+	if consumedQuery != "" {
+		logger.Trace("Executing consumed query", watermill.LogFields{
```

Comment:
"consumed query" isn't very explicit about what it is supposed to do.
it's in the trace level, so it's not a big problem, but maybe:
`Executing query to confirm message consumed` or sth similar?

Reply:
done